mindsdb/interfaces/storage/fs.py

```diff
@@ -393,53 +421,44 @@ def __init__(self, resource_group: str, resource_id: int,
             self.folder_path.mkdir(parents=True, exist_ok=True)
 
     @profiler.profile()
-    def push(self, compression_level=9):
-        with FileLock(self.folder_path):
+    def push(self, compression_level: int = 9):
+        with FileLock(self.folder_path, mode='r'):
             self._push_no_lock(compression_level=compression_level)
 
     @profiler.profile()
-    def _push_no_lock(self, compression_level=9):
+    def _push_no_lock(self, compression_level: int = 9):
         self.fs_store.put(
             str(self.folder_name),
             str(self.resource_group_path),
             compression_level=compression_level
         )
 
     @profiler.profile()
-    def push_path(self, path):
-        with FileLock(self.folder_path):
-            self.fs_store.put(os.path.join(self.folder_name, path), str(self.resource_group_path))
+    def push_path(self, path, compression_level: int = 9):
+        # TODO implement push per element
+        self.push(compression_level=compression_level)
 
     @profiler.profile()
     def pull(self):
-        with FileLock(self.folder_path):
-            self._pull_no_lock()
-
-    @profiler.profile()
-    def _pull_no_lock(self):
         try:
-            self.fs_store.get(str(self.folder_name), str(self.resource_group_path))
-        except Exception:
+            self.fs_store.get(
+                str(self.folder_name),
+                str(self.resource_group_path)
+            )
+        except (FileNotFoundError, S3ClientError):
             pass
 
     @profiler.profile()
-    def pull_path(self, path, update=True):
-        with FileLock(self.folder_path):
-            if update is False:
-                # not pull from source if object is exist
-                if os.path.exists(self.resource_group_path / self.folder_name / path):
-                    return
-            try:
-                # TODO not sync if not changed?
-                self.fs_store.get(os.path.join(self.folder_name, path), str(self.resource_group_path))
-            except Exception:
-                pass
+    def pull_path(self, path):
+        # TODO implement pull per element
```

Comment:
will it be implemented before merge? because path is used now

Reply:
no, this method should not exists, because we may copy only whole base folder if use s3 as storage